doc/manual/metaprogramming.rst

```diff
@@ -134,19 +167,102 @@ ambiguity in parsing.:
     julia> :(::)
     :(::)
 
-:obj:`Symbol`\ s can also be created using :func:`symbol`, which takes
-a character or string as its argument:
+Expressions and evaluation
+--------------------------
+
+Quoting
+~~~~~~~
+
+The second syntactic purpose of the ``:`` character is to create expression
+objects without using the explicit :obj:`Expr` constructor. This is referred
+to as *quoting*. The ``:`` character, followed by a parenthesized Julia
+statement, generates an :obj:`Expr` object based on the enclosed code.
+Here is example of the short form used to quote an arithmetic expression:
 
 .. doctest::
 
-    julia> symbol('\'')
-    :'
+    julia> ex = :(a+b*c+1)
+    :(a + b * c + 1)
 
-    julia> symbol("'")
-    :'
+    julia> typeof(ex)
+    Expr
+
+(to view the structure of this expression, try ``ex.head`` and ``ex.args``,
+or use :func:`dump` as above).
 
-:func:`eval` and Interpolation
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Note that equivalent expressions may be constructed using :func:`parse` or
+the direct :obj:`Expr` form:
+
+.. doctest::
+
+   julia>      :(a + b*c + 1)  ==
+     parse("a + b*c + 1") ==
+	 Expr(:call, :+, :a, Expr(:call, :*, :b, :c), 1)
+   true
+
+Expressions provided by the parser generally only have symbols, other
+expressions, and literal values as their args, whereas expressions
+constructed by Julia code can have arbitrary run-time values
+without literal forms as args. In this specific example, ``+`` and ``a``
+are symbols, ``*(b,c)`` is a subexpression, and ``1`` is a literal
+64-bit signed integer.
+
+There is a second syntactic form of quoting for multiple expressions:
+blocks of code enclosed in ``quote ... end``. Note that this form
+introduces :obj:`QuoteNode` elements to the expression tree, which
+must be considered when directly manipulating an expression tree
+generated from ``quote`` blocks. For other purposes, ``:( ... )``
+and ``quote .. end`` blocks are treated identically.
+
+.. doctest::
+
+    julia> ex = quote
+		    x = 1
+		    y = 2
+		    x + y
+		end
+    quote  # none, line 2:
+	x = 1 # line 3:
+	y = 2 # line 4:
+	x + y
+    end
+
+    julia> typeof(ex)
+    Expr
+
+Interpolation
+~~~~~~~~~~~~~
+
+Direct construction of :obj:`Expr` objects with value arguments is
+powerful, but :obj:`Expr` constructors can be tedious compared to "normal"
+Julia syntax. As an alternative, Julia allows "splicing" or interpolation
+of literals or expressions into quoted expressions. Interpolation is
+indicated by the ``$`` prefix:
+
+.. doctest::
+
+    julia> a = 1;
+
+    julia> ex = :($a + b)  # interpolate the literal value of `a`
+    :(1 + b)
+
+    julia> ex = :(a in $:((1,2,3)) )
+    :($(Expr(:in, :a, :((1,2,3)))))
+
+Interpolating symbols into a nested expression requires enclosing each 
+symbol in an enclosing quote block:
+
+    julia> :( :a in $( :(:a + :b) ) )
```

Comment:
Maybe put a series of `^^^^^` instead of just one at the beginning and one at the end.